02_knight_and_princess/novikov_dmitry/graph.cpp

```diff
@@ -0,0 +1,238 @@
+#include "graph.hpp"
+
+namespace uni_cpp_practice {
+bool check_gray_valid(const Vertex& first_vertex, const Vertex& second_vertex) {
+  if (first_vertex.get_edge_ids().size() == 0 ||
+      second_vertex.get_edge_ids().size() == 0) {  //только текущее ребро
+    return true;
+  }
+  return false;
+}
+
+bool check_green_valid(const Vertex& first_vertex,
+                       const Vertex& second_vertex) {
+  if (first_vertex.depth == second_vertex.depth) {
+    return true;
+  }
+  return false;
+}
+
+bool check_blue_valid(const Vertex& first_vertex, const Vertex& second_vertex) {
+  if (first_vertex.depth == second_vertex.depth) {
+    if (std::abs(first_vertex.id - second_vertex.id) == 1) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool check_yellow_valid(const Vertex& first_vertex,
+                        const Vertex& second_vertex) {
+  if (std::abs(first_vertex.depth - second_vertex.depth) == 1) {
+    return true;
+  }
+  return false;
+}
+
+bool check_red_valid(const Vertex& first_vertex, const Vertex& second_vertex) {
+  if (std::abs(first_vertex.depth - second_vertex.depth) == 2) {
+    return true;
+  }
+  return false;
+}
+
+bool check_color_valid(const Vertex& first_vertex,

```

Comment:
Все-таки это должно быть в приватном неймспейсе, попробуй добавить сюда `[[maybe_unused]]`
```suggestion
[[maybe_unused]] bool check_color_valid(const Vertex& first_vertex,
```